### DISCUSSION TITLE
impl Ignorable for Vec?

### DISCUSSION THREAD

**Discussion by timmc** [rwf2/Rocket#2883] [Ideas]
Would it make sense to make `Vec` implement `Ignorable`? Right now only `Option` and `Result` are `Ignorable`: https://api.rocket.rs/v0.5/rocket/http/uri/fmt/trait.Ignorable

It would be great if I could pass `uri!(my_route(things = _))` when `things: Vec<_>` -- especially because I can't just use `vec![]`; I need to specify type annotations: `uri!(my_route(things = Vec::<Thing>::new()))`. It's kind of annoying and doesn't let me easily read it as "oh, I just want the default value here".

For that matter, should there be a blanket implementation for `Default`?

**Comment by the10thWiz** [comment]
I think this does make sense for `Vec<T>`, since the default would be empty.

I don't think it makes sense as a blanket implementation for `Default`, since it's more complex than that. What it's supposed to express is that the field will be omitted by the `uri!` macro - e.g. the following code snippet would produce a uri of `/item/100?`.

```rust
#[get("/item/<id>?<track>")]
fn get_item(id: i32, track: Option<u8>) { /* .. */ }

uri!(get_item(id = 100, track = _));
```

A PR to add a `Ignorable` impl for `Vec<T>` would be accepted.
